src/clients.rs

```diff
@@ -100,12 +85,12 @@ impl ProverClient for Prover {
         let msg = ExecuteProverRequest { utx: utx_bytes };
         let req = tonic::Request::new(msg);
 
-        self.status("Proving tx, please wait...");
+        (self.status)("Proving tx, please wait...");
         let mut prover = self.client.lock().unwrap();
         let proof_bytes = prover.prove_execute(req).wait()?.into_inner().proof;
-        self.status("Proof success!");
+        (self.status)("Proof success!");
```

Comment:
If you add a method in the `Prover` and `State` like this:

```rust
impl Prover {
    fn status(&self, text: &str) {
        (self.status)(text)
    }
}
```

You can remove the brackets from all the calls, since you're not calling a property anymore but a method.

Reply:
Sure, let's do it!